### PR TITLE
phpunit update

### PR DESCRIPTION
**Pull request by kokspflanze**
- PHPUnit_Framework_TestCase changed to PHPUnit\Framework\TestCase
- getMock changed to createMock
- updated min phpunit version
- updated composer.lock, with latest phpunit version
- updated travis build, with phpunit as build param

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -68,15 +70,17 @@ before_install:
...
-  - composer show --installed
+  - if [[ $TEST_COVERAGE == 'true' ]]; then travis_retry composer require --dev $COMPOSER_ARGS $COVERAGE_DEPS ; fi
+  - composer show
 
 script:
-  - if [[ $TEST_COVERAGE == 'true' ]]; then composer test-coverage ; else composer test ; fi
-  - if [[ $CHECK_CS == 'true' ]]; then composer cs-check ; fi
+  - if [[ $TEST_COVERAGE == 'true' ]]; then ./vendor/bin/phpunit --coverage-clover clover.xml ; fi
+  - if [[ $TEST_COVERAGE != 'true' ]]; then ./vendor/bin/phpunit ; fi
+  - if [[ $CS_CHECK == 'true' ]]; then ./vendor/bin/php-cs-fixer fix -v --diff --dry-run ; fi
```
Comment (michalbundyra) [inline .travis.yml]: Why these lines are changed? Why you are not using composer scripts defined in `composer.json` ?

File: .travis.yml [hunk 2]
```diff
@@ -68,15 +70,17 @@ before_install:
...
 install:
+  - travis_retry composer install $COMPOSER_ARGS
+  - if [[ $TRAVIS_PHP_VERSION =~ ^5.6 ]]; then travis_retry composer update $COMPOSER_ARGS --with-dependencies $LEGACY_DEPS ; fi
   - if [[ $DEPS == 'latest' ]]; then travis_retry composer update $COMPOSER_ARGS ; fi
   - if [[ $DEPS == 'lowest' ]]; then travis_retry composer update --prefer-lowest --prefer-stable $COMPOSER_ARGS ; fi
-  - if [[ $TEST_COVERAGE == 'true' ]]; then travis_retry composer require --no-update --dev $COMPOSER_ARGS satooshi/php-coveralls ; fi
-  - travis_retry composer install $COMPOSER_ARGS
-  - composer show --installed
+  - if [[ $TEST_COVERAGE == 'true' ]]; then travis_retry composer require --dev $COMPOSER_ARGS $COVERAGE_DEPS ; fi
+  - composer show
```
Comment (michalbundyra) [inline .travis.yml]: We should have here `stty cols 120 && composer show`

**Review by michalbundyra — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (kokspflanze) [reply .travis.yml]: i rechanged i, with my latest commit to the [template](https://github.com/zendframework/maintainers/blob/master/template/.travis.yml) @froschdesign gave me the link

File: .travis.yml (hunk 2 above)
Comment (kokspflanze) [reply .travis.yml]: changed to the template

File: composer.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
         "zendframework/zend-validator": "^2.6"
     },
     "require-dev": {
-        "phpunit/phpunit": "^4.5",
+        "phpunit/phpunit": "^5.7.21 || ^6.3.0",
```
Comment (michalbundyra) [inline composer.json]: It's not too important, but we don't need `.0` in `6.3.0`, please note we have it like that for other packages, for example:
`"zendframework/zend-validator": "^2.6"`
not:
`"zendframework/zend-validator": "^2.6.0"`

File: .travis.yml [hunk 4]
```diff
@@ -2,48 +2,34 @@ sudo: false
...
 env:
   global:
-    - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
-    - SITE_URL=https://zendframework.github.io/zend-mvc-i18n
-    - GH_USER_NAME="Matthew Weier O'Phinney"
-    - GH_USER_EMAIL=matthew@weierophinney.net
-    - GH_REF=github.com/zendframework/zend-mvc-i18n.git
-    - secure: "eZcLCwjTae1kWFL9dlQ4wts9SKW1bgvNYQn2zk7oeu0cWwLDDlMxoKg3ko0UcWDfLWi9kQetpoPvkAIrxsMhDwnzdHQ4e83GbvF4XqoB6pxr1T29yjqbfY7Kd6Y7CErE1Scfz25gaoewLjW09/fiXiqHTECnZTkwELwQy6LvOmE+eYbPRLHGyEE/9CVhyjcPCH9DQ1Kj/dZG2dLFjbV1kkJXkp/6m4zNIOLsDE6Y2tVhkgeVeIir3gpJzfkghT/O/P/93KLbCPcI+DVmbYi15VsUso2QGjlJFkhcvU0B/SxM4rsPUVWfgavOdhpW0vdEiLIWqWpxJ4AZL8VAx3Ib8b8xjWNHB95JSSgldoQDitvtM/S+PKlT1T1Mc4r/waAyXkZIcmrXlCjYjvMnIw7IXCr6geNNFwOyZeNPHgdBcKYQyZOH0dz2NGFcol195AxrTpQqQVuYFRDYO2p52q4qSdIBIPr08Rv4vWT2G5kXCQbzrihmQXEj7zRwEofHxJWVlfS3Fo6eqS9zOAwU+IboyQ8Nh3Uddqc7szFJB+IBr8YM1kOhVs8FdYo6SvEHPMtutwE6+rdQMQ+m5v4z1JuW+HuhpPxWwm8SE5tqer9HN+KuqbgcnycOM5sHo7FGntBea3L7x7EE8yfkUmChatMMhMghnZ3WgwULsKjbrD97bE8="
+    - COMPOSER_ARGS="--no-interaction"
+    - COVERAGE_DEPS="satooshi/php-coveralls"
```
Comment (michalbundyra) [inline .travis.yml:12]: We need legacy dependencies here.

**Review by michalbundyra — CHANGES_REQUESTED** [review]
I think we need here still modifications in Travis configuration to work with legacy dependencies. Currently library supports PHP 5.6 and PHP 7+.  As far as I know `travis.yml` from maintainers template is for libraries which supports only PHP 7.1+

File: .travis.yml (hunk 4 above)
Comment (kokspflanze) [reply .travis.yml:12]: im working on it but it will take some time, i have many open prs, with that old case

**Pull request merged**